js/tji-datasets-explore.js

```diff
@@ -1,15 +1,349 @@
+// *******************************************************************
 
-//TOGGLE CHECKBOXES
-jQuery(document).ready(function($) {
-  $("#filter-section input:checkbox").prop("checked", true);
 
-  $('.js-filter-input-all').click(function() {
-      var $checkboxes = $(this).parent().parent().parent().find('input[type=checkbox]');
-      $checkboxes.prop('checked', $(this).is(':checked'));
-  }); 
-});
\ No newline at end of file
+var TJIGroupByBarChart = function(elt_id, groupBy, data, missing_data_label) {
+  this.elt_id = elt_id;
+  this.groupBy = groupBy;
+  this.chart = null;
+  this.missing_data_label = missing_data_label || '(not given)';
+  this.color_mapping = null;
+  this.create(data);
+}
+
+TJIGroupByBarChart.prototype.type = 'bar';
+
+TJIGroupByBarChart.prototype.color_palette = [COLOR_TJI_BLUE];
+
+TJIGroupByBarChart.prototype.create = function(data) {
+  var that = this;
+  var grouped = this.get_group_counts(data);
+
+  // 'create' is only run with the full data set.
+  // So we store our color mapping (e.g. "Male" = blue) so that
+  // subsequent filterings and changes don't alter the color
+  // associated with males.
+  this.color_mapping = {};
+  _.each(grouped.keys, function(k, idx) {
+    that.color_mapping[k] = that.color_palette[idx % that.color_palette.length];
+  });
+  var colors = _.map(grouped.keys, function(k) {
+    return that.color_mapping[k]
+  })
+
+  // Build the chart
+  this.chart = new Chart(document.getElementById(this.elt_id).getContext('2d'), {
+    type: this.type,
+    data: {
+      labels: grouped.keys,
+      datasets:[
+        {
+          data: grouped.counts,
+          fill: false,
+          backgroundColor: colors,
+          lineTension: 0.1
+        }
+      ]
+    },
+    options: this.get_options()
+  });
+}
+
+TJIGroupByBarChart.prototype.update = function(data) {
+  var that = this;
+  var grouped = this.get_group_counts(data);
+  this.chart.data.datasets[0].data = grouped.counts;
+  this.chart.data.labels = grouped.keys;
+  
+  this.chart.data.datasets[0].backgroundColor = _.map(grouped.keys, function(k) {
+    return that.color_mapping[k]
+  });
+  this.chart.update();
+}
+
+TJIGroupByBarChart.prototype.get_options = function() {
+  var options = {
+    title: {
+      display: true,
+      text: "By " + this.groupBy.replace(/_/g, " "),  // Convert underscores to spaces
+      fontSize: 36,
+    },
+    legend: {
+      display: false
+    },
+    scales: {
+      yAxes: [{
+        ticks: {
+          beginAtZero:true
+        }
+      }]
+    }
+  }
+  return _.extend(options, this.get_options_overrides());
+}
+
+TJIGroupByBarChart.prototype.get_options_overrides = function() {
+  return {};
+}
+
+TJIGroupByBarChart.prototype.get_group_counts = function(data) {
+  data = _.filter(data, this.groupBy);
+  var grouped = _.groupBy(data, this.groupBy);
+  var keys = _.sortBy(_.keys(grouped));
+  if (keys.indexOf(this.missing_data_label) !== -1) {
```

Comment:
can we add a comment above 
//moving 'missing data group' to last position
I liked that you were commenting in places where the code was extra complex to make code more readable.